module-4/backend/week_one.md

```diff
@@ -5,18 +5,43 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+Mocha at the least, and Chai for better asserts.
 4. What is the syntax for invoking a function?
+```javascript
+function() <- needs the parantheses, function just returns the actual function, which is useful for sending callbacks etc...
+```
 5. What is CORS?
+Cross-Origin Resource Sharing. My understanding is that it allows one web origin to access data from another web origin.
 6. How do we enable CORS?
+I have no idea. Yet.
 7. What's `this` in JavaScript?
+This is similar to 'self' in Ruby, but with some intersting scoping differences. In a nutshell, it is a reference to the object 'this' is wrapped in. Being inside a for loop or forEach method, for example, 'this' will reference the loop. This starts to get weird inside of a class. If you're in declared method in the class this references the entire class. As soon as you enter another function nested inside of that fuction 'this' will now reference that function. You can bind 'this' to a function by attatching {}.bind(this) to the end of it. This will scope 'this' to the object directly outside of the function.
 8. What is Webpack and why is it useful?
+Webpack is a module bundler, sort of like bundler for ruby, and will build your assets for you. But it's so much more! It also allows you to run a server and make HTML calls in a test environment. I suspect it'll even more useful later.
 9. When/why do you want to use event delegation?
+I'm still a little fuzzy on this, but what I'm trying to do is this: A add a bunch of rows to a table. Each row can be deleted or edited. Instead of adding event listeners to each row I can add an event listener to the table that watches for a row being clicked. Since the call bubbles down (or up. whatever), The table can listen for an event that matches a row triggaring other things to happen.
 10. What is a callback function?
+A callback function is a function that can be sent to another function to be called when the first function does it's thing. This seems to get a little wierd when making asych calls, but I'm getting there.
 11. What's `npm` and what do we use it for?
+Nutritious Puffy Marshmallows. Packages in node are like gems in ruby. NPM can bundle up and install packages and dependencies for your app based on a package.json file.
 
 #### Review  
 12. What's the MVC design pattern? Describe each part of MVC.
+Model, View, Controller.
+The model accesses your database or acts as a non-database object or PORO (plain ol' ruby object).
+The view renders your html. It will need access to objects to render dynamic pages, but 99% of logic should be kept out.
+The controller actually does a few things. It acts as a crossroads for your RESTful routes, but then also collects data for access by the view. The access data part _should_ be minimal. If a lot of data needs to be collected that responsibility should be moves to a presenter.
+
 13. What are a few ways to optimize a Rails application?
+- Don't write it in Rails.
```

Comment:
I'd extrapolate your suggestion to not use Rails (it can be a serious suggestion, just back it up)